lib/formatters/json-with-metadata.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * @fileoverview JSON reporter
+ * @author Burak Yigit Kaya aka BYK
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+module.exports = function(results, data) {
+    return JSON.stringify({
+        results,
+        rulesMeta: data.rulesMeta
```

Comment:
Could this just expose a `data` property? It seems like that would make the properties of the object consistent with the arguments provided to the formatter, which could be mildly useful if we add more properties to `data` in the future.

```suggestion
        data
```

Reply:
Yes this is better. No change needed if something new is added to the data object.